src/streamlink/plugins/nbcnews.py

```diff
@@ -21,23 +22,30 @@ class NBCNews(Plugin):
     json_data_re = re.compile(
         r'<script type="application/ld\+json">({.*?})</script>'
     )
-    api_url = 'https://stream.nbcnews.com/data/live_sources_{}.json'
+    api_url = 'https://api-leap.nbcsports.com/feeds/assets/{}?application=NBCNews&format=nbc-player&platform=desktop'
     token_url = 'https://tokens.playmakerservices.com/'
 
     api_schema = validate.Schema(
-        validate.parse_json(),
+        validate.transform(parse_json),
         {
             'videoSources': [{
-                'sourceUrl': validate.url(),
-                'type': validate.text,
+                'cdnSources': {
+                    'primary': [{
+                        'sourceUrl': validate.url(),
+                    }]
+                }
             }],
         },
         validate.get('videoSources'),
         validate.get(0),
+        validate.get('cdnSources'),
+        validate.get('primary'),
+        validate.get(0),
+        validate.get('sourceUrl'),
```

Comment:
`validate.get` can receive a tuple of strings which will call `__getitem__` on each successive object, which is much better than calling `validate.get` lots of times.